core/src/main/scala/io/projectglow/vcf/VCFHeaderUtils.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.vcf
+
+import java.io.StringReader
+import java.net.{URI, URISyntaxException}
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+import com.google.common.annotations.VisibleForTesting
+import htsjdk.variant.vcf.{VCFCodec, VCFHeader, VCFHeaderLine}
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.sql.types.StructType
+
+import io.projectglow.common.GlowLogging
+
+object VCFHeaderUtils extends GlowLogging {
+
+  val VCF_HEADER_KEY = "vcfHeader"
+  val INFER_HEADER = "infer"
+
+  def parseHeaderFromString(s: String): VCFHeader = {
+    val stringReader = new StringReader(s)
+    val lineIterator = new LineIteratorImpl(IOUtils.lineIterator(stringReader).asScala)
+    val codec = new VCFCodec()
+    try {
+      codec.readActualHeader(lineIterator).asInstanceOf[VCFHeader]
+    } catch {
+      case NonFatal(e) =>
+        logger.warn(s"Wasn't able to parse VCF header in $s. $e")
+        new VCFHeader()
```

Comment:
I can't imagine that this is the correct error behavior... shouldn't we throw this exception?

Reply:
Huh, I don't remember why we did this. I'll change it.